lib/app.rb

```diff
@@ -153,4 +150,33 @@ def run
       command = @input_stream.gets.chomp
     end
   end
+
+  private
+
+  def current_project?
+    !!current_project
+  end
+
+  def get_project_name
+    prompt_for_project
+    get_user_input
+  end
+
+  def prompt_for_project
+    output_stream.puts("\e[0;35mEnter a project name:\e[0m")
+  end
+
+  def switch_project
+    @current_project = projects.detect { |p|
+      p.keys.first == command_or_project_name
+    }
+  end
+
+  def get_user_input
+    @command_or_project_name = input_stream.gets.chomp
```

Comment:
Went extra verbose with this name so I don't forget that it's being used for multiple things.